src/main/scala/mesosphere/marathon/experimental/repository/TemplateRepositoryLike.scala

```diff
@@ -0,0 +1,115 @@
+package mesosphere.marathon
+package experimental.repository
+
+import akka.Done
+import mesosphere.marathon.state.PathId
+
+import scala.concurrent.Future
+import scala.util.Try
+
+trait TemplateRepositoryLike {
```

Comment:
Since we're planning on merging pods apps in the new template repository, maybe it would be better to create an interim type PodOrSpec instead of introducing this Generic parameter?

Reply:
We're planning to introduce a new type (a real template) which will combine both. It might happen that we'll introduce an archetype for all of them but for now, the easiest way was to duck-type them together.